### PR TITLE
Adds createPeginV1TxData function

### PR DESCRIPTION
**Pull request by jeremy-then**
Changes:

1) Added a new `createPeginV1TxData` function easily create the pegin v1 data for an rsk recipient address and optionally a btc refund address.

2) Refactored the `src/pegin-address-verifier.js` file to be able to call one function from another.

3) Added tests for the new `createPeginV1TxData` function.

### CODE REVIEW

File: test/pegin-address-verifier.js [hunk 1]
```diff
@@ -128,3 +128,46 @@ describe('decodeWifPrivateKey tests', () => {
...
+
+    it('should throw error if btc refund address is not P2PKH or P2SH', () => {
+        const rskAddress = '0xa65C9bF91Ae59fC5bE36300D0BC271998529bb75';
+        const btcAddress = 'bc1qar0srrr7xfkvy5l643lydnw9re59gtzzwf5mdq';
+        expect(() => validator.createPeginV1TxData(rskAddress, btcAddress)).to.throw('Unsupported btc refund address type: bech32');
+    });
+
+    it('should throw error if rsk address is not provided', () => {
+        expect(() => validator.createPeginV1TxData()).to.throw('RSK destination address is required');
+    });
```
Comment (marcos-iov) [inline test/pegin-address-verifier.js]: Repeated

File: test/pegin-address-verifier.js [hunk 2]
```diff
@@ -128,3 +128,46 @@ describe('decodeWifPrivateKey tests', () => {
...
+        expect(validator.createPeginV1TxData(rskAddress)).to.be.eq(expectedData);
+    });
+
+    it('should create pegin v1 data with rsk address without the 0x prefix', () => {
+        const rskAddress = 'a65C9bF91Ae59fC5bE36300D0BC271998529bb75';
+        const expectedData = '52534b5401a65C9bF91Ae59fC5bE36300D0BC271998529bb75';
+        expect(validator.createPeginV1TxData(rskAddress)).to.be.eq(expectedData);
+    });
+
+    it('should create pegin v1 data with btc refund address', () => {
```
Comment (marcos-iov) [inline test/pegin-address-verifier.js]: one with a p2sh address?

File: test/pegin-address-verifier.js [hunk 3]
```diff
@@ -128,3 +128,46 @@ describe('decodeWifPrivateKey tests', () => {
...
+
+    it('should create pegin v1 data without btc refund address', () => {
+        const rskAddress = '0xa65C9bF91Ae59fC5bE36300D0BC271998529bb75';
+        const expectedData = '52534b5401a65C9bF91Ae59fC5bE36300D0BC271998529bb75';
+        expect(validator.createPeginV1TxData(rskAddress)).to.be.eq(expectedData);
+    });
+
+    it('should create pegin v1 data with rsk address without the 0x prefix', () => {
+        const rskAddress = 'a65C9bF91Ae59fC5bE36300D0BC271998529bb75';
+        const expectedData = '52534b5401a65C9bF91Ae59fC5bE36300D0BC271998529bb75';
```
Comment (marcos-iov) [inline test/pegin-address-verifier.js]: should the expected data be in lower case or it doesn't make a difference?

**Review by marcos-iov — COMMENTED** [review]
Well done Jere, just a couple of comments in the tests.

Let's also bump the version number and recreate package-lock file using node v20

File: test/pegin-address-verifier.js (hunk 1 above)
Comment (jeremy-then) [reply test/pegin-address-verifier.js]: Removed!

File: test/pegin-address-verifier.js (hunk 2 above)
Comment (jeremy-then) [reply test/pegin-address-verifier.js]: Done!

File: test/pegin-address-verifier.js (hunk 3 above)
Comment (jeremy-then) [reply test/pegin-address-verifier.js]: It doesn't make a difference. It's hex data. But as a standard now I'm returning it in lowercase.

**Review by marcos-iov — APPROVED** [review]
Well done!

File: src/crypto/constants.js [hunk 4]
```diff
@@ -8,15 +8,21 @@ const NETWORKS = {
...
 const HASH_FIELD_NAMES = {
     p2pkh: 'scriptPubKey',
     p2sh: 'scriptHash'
-}
+};
+
+const ADDRESS_TYPES_CODES = {
+    p2pkh: '01',
+    p2sh: '02'
+};
```
Comment (nathanieliov) [inline src/crypto/constants.js:22]:
```suggestion
}; // Check https://github.com/rsksmart/RSKIPs/blob/2c994cc108885ccc5a116e4aee8c073b5eca5682/IPs/RSKIP170.md form ore details.
```

File: src/pegin-address-verifier.js [hunk 5]
```diff
@@ -1,19 +1,71 @@
...
+            data += ADDRESS_TYPES_CODES[refundAddressInfo.type];
+            switch (refundAddressInfo.type) {
+                case 'p2pkh':
+                    data += refundAddressInfo.scriptPubKey;
+                    break;
+                case 'p2sh':
+                    data += refundAddressInfo.scriptHash;
+                    break;
+                default:
+                    throw new Error(`Unsupported btc refund address type: ${refundAddressInfo.type}`);
```
Comment (nathanieliov) [inline src/pegin-address-verifier.js]:
```suggestion
            switch (refundAddressInfo.type) {
                case 'p2pkh':
                    data += ADDRESS_TYPES_CODES. p2pkh + refundAddressInfo.scriptPubKey;
                    break;
                case 'p2sh':
                    data += ADDRESS_TYPES_CODES. p2sh + refundAddressInfo.scriptHash;
                    break;
                default:
                    throw new Error(`Unsupported btc refund address type: ${refundAddressInfo.type}`);
```

File: src/pegin-address-verifier.js [hunk 6]
```diff
@@ -1,19 +1,71 @@
...
+    if (rskDestinationAddress.startsWith('0x')) {
+        rskDestinationAddress = rskDestinationAddress.substring(2);
+    }
+
+    data += rskDestinationAddress;
+
+    if (btcRefundAddress) {
+        const refundAddressInfo = getAddressInformation(btcRefundAddress);
+        if (refundAddressInfo) {
+            data += ADDRESS_TYPES_CODES[refundAddressInfo.type];
```
Comment (nathanieliov) [inline src/pegin-address-verifier.js]: What happened if the type is not found in the ADDRESS_TYPES_CODES?

I prefer to move this below to the case code block where you know what type is the refund address.

**Review by nathanieliov — COMMENTED** [review]
just a few suggestions

File: src/pegin-address-verifier.js (hunk 6 above)
Comment (jeremy-then) [reply src/pegin-address-verifier.js]: It would throw an exception.

File: src/pegin-address-verifier.js (hunk 5 above)
Comment (jeremy-then) [reply src/pegin-address-verifier.js]: Updated!

File: src/crypto/constants.js (hunk 4 above)
Comment (jeremy-then) [reply src/crypto/constants.js:22]: Updated!

**Review by nathanieliov — APPROVED** [review]
LGTM

**Pull request merged**